### PR TITLE
Best practices for update_actions() method

### PR DESCRIPTION
**Pull request by kgaillot**
This continues bringing libpacemaker in line with current development guidelines, focusing on the resource_alloc_functions_s:update_actions() method.

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , here's another low-priority review when you get a chance. Thanks

File: lib/pacemaker/pcmk_sched_actions.c [hunk 1]
```diff
@@ -696,6 +696,56 @@ is_primitive_action(pe_action_t *action)
...
+         */
+        return;
+    }
+
+    if ((then_rsc_role >= RSC_ROLE_STARTED)
+        && pcmk_is_set(then->flags, pe_action_optional)
+        && (then->node != NULL)
+        && pcmk__list_of_1(then_on)
+        && (then->node->details == ((pe_node_t *) then_on->data)->details)
+        && pcmk__str_eq(then->task, RSC_START, pcmk__str_none)) {
```
Comment (clumens) [inline lib/pacemaker/pcmk_sched_actions.c:733]: Did you intend to change this comparison and the one above to pcmk__str_none?

Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_actions.c:733]: Yes, I've been doing those as I run across them. Probably should be separate commits but I never remember to do that ...

Comment (clumens) [reply lib/pacemaker/pcmk_sched_actions.c:733]: Looks good to me, then.

**Pull request merged**